packages/esm-patient-appointments-app/src/hooks/useAppointments.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState, useMemo } from 'react';
+import dayjs from 'dayjs';
+import { getAppointments } from '../appointments/appointments.resource';
+import { Appointment } from '../types';
+
+export const useAppointments = (patientUuid: string) => {
+  const [patientAppointments, setPatientAppointments] = useState<Array<Appointment>>([]);
+  const [error, setError] = useState<Error>(null);
+  const [status, setStatus] = useState<'pending' | 'resolved' | 'error'>('pending');
+
+  useEffect(() => {
+    if (patientUuid) {
+      const ac = new AbortController();
+      const startDate = dayjs(new Date().toISOString()).subtract(6, 'month').toISOString();
+      getAppointments(patientUuid, startDate, ac).then(
+        ({ data }) => {
+          setPatientAppointments(data);
```

Comment:
That multiple set statements are used in a row here is an indicator of a coupled state. This should be replaced either by a more complex object as state or by a `useReducer` that drives setting that more complex state.